internal/installercache/installercache.go

```diff
@@ -138,14 +140,25 @@ func (i *Installers) Get(ctx context.Context, releaseID, releaseIDMirror, pullSe
 	for {
 		select {
 		case <-ctx.Done():
-			return nil, ctx.Err()
+			err := ctx.Err()
+			if err == context.DeadlineExceeded {
+				i.metricsAPI.InstallerCacheGetReleaseTimeout(releaseID)
+			}
+			return nil, err
 		default:
-			release, err := i.get(releaseID, releaseIDMirror, pullSecret, ocRelease, ocpVersion, clusterID)
+			majorMinorVersion, err := ocRelease.GetMajorMinorVersion(i.log, releaseID, releaseIDMirror, pullSecret)
```

Comment:
I think this should be done outside the for select

Reply:
I don't see any point to doing that as nothing apart from this case requires this.
But since you asked, I'll move it...